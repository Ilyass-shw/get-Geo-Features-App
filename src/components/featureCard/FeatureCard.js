import LazyLoad from "react-lazyload";
import ReactJson from "react-json-view";
import { Card } from "./FeatureCardStyles";

const FeatureCard = ({ feature }) => {
  return (
    <LazyLoad height={30} offset={150}>
      <Card>
        <ReactJson src={feature} />
      </Card>
    </LazyLoad>
  );
};

export default FeatureCard;
