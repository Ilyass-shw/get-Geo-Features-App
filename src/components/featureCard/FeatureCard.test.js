import { render } from "@testing-library/react";
import FeatureCard from "./FeatureCard";

describe("FeaturesCard", () => {
  const feature = {
    type: "Feature",
    id: "node/7335291821",
    properties: {
      timestamp: "2020-03-27T15:43:07Z",
      version: "1",
      changeset: "82722505",
      user: "gislars",
      uid: "108133",
      amenity: "icycle_parking",
      bicycle_parking: "stands",
      capacity: "4",
      covered: "no",
      id: "node/7335291821",
    },
    geometry: {
      type: "Point",
      coordinates: ["13.424746", "52.5333751"],
    },
  };

  it("should render", () => {
    render(<FeatureCard feature={feature} />);
  });
  it("should match snapshot", () => {
    const { asFragment } = render(<FeatureCard feature={feature} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
