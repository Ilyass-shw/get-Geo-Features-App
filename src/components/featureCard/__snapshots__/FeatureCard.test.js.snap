// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturesCard should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="lazyload-wrapper "
  >
    <div
      class="lazyload-placeholder"
      style="height: 30px;"
    />
  </div>
</DocumentFragment>
`;
